src/c_aci_testing/tools/aci_deploy.py

```diff
@@ -58,7 +67,12 @@ def aci_deploy(
     ]))
     print("")
 
-    subprocess.run(az_command, check=True)
+    sys.stdout.flush()
+    sys.stderr.flush()
+
+    res = subprocess.run(az_command)
+    if res.returncode != 0:
+        raise Exception(f"Deployment failed with return code {res.returncode}")
```

Comment:
nit: Can we use more specific exception? Ideally some custom error like DeploymentError or built-in type like RuntimeError which is a bit more specific than `Exception` class

Reply:
Since nothing is actually catching those exceptions (this whole thing is currently mostly invoked from shell scripts - I don't think anything in the dashboard uses `target_run`) I probably wouldn't go as far as having a custom exception, but RuntimeError might be good

Reply:
That sounds reasonable to me